File: findMaxConsecutiveOnes.js
// Given a binary array, find the maximum number of consecutive 1s in this array.
// Time: O(n)
// Space: O(1)

/**
 * @param {number[]} nums
 * @return {number}
 */

const nums = [1, 0, 1, 1, 0, 1];
console.log(findMaxConsecutiveOnes(nums));

function findMaxConsecutiveOnes(nums) {
    let count = 0;
    let maxCount = 0;
    for (num of nums) {
        if (num == 1)
            count++;
        else {
            maxCount = Math.max(maxCount, count);
            count = 0;
        }
    }

    return Math.max(maxCount, count);
};

File: findMaxConsecutiveOnes.py
# Given a binary array, find the maximum number of consecutive 1s in this array.
# Time: O(n)
# Space: O(1)

def findMaxConsecutiveOnes(nums):
    count = 0
    max_count = 0
    for num in nums:
        if num == 1:
            count += 1
        else:
            max_count = max(max_count, count)
            count = 0

    return max(max_count, count)


nums = [1, 0, 1, 1, 0, 1]
print(findMaxConsecutiveOnes(nums))
